src/Core/Authorization/AuthorizationResolver.cs

```diff
@@ -479,19 +494,23 @@ public IEnumerable<string> GetAllowedExposedColumns(string entityName, string ro
                  * claim.Value: "authz@microsoft.com"
                  * claim.ValueType: "string"
                  */
-                // At this point, only add non-role claims to the collection and only throw an exception for duplicate non-role claims.
+                // Ignore role claims because they have already been processed.
+                // Only the role claim designated in the client role header is added to the claimsInRequestContext dictionary.
                 if (!claim.Type.Equals(AuthenticationOptions.ROLE_CLAIM_TYPE) && !claimsInRequestContext.TryAdd(claim.Type, claim))
                 {
-                    // If there are duplicate claims present in the request, return an exception.
-                    throw new DataApiBuilderException(
-                        message: "Duplicate claims are not allowed within a request.",
-                        statusCode: HttpStatusCode.Forbidden,
-                        subStatusCode: DataApiBuilderException.SubStatusCodes.AuthorizationCheckFailed
-                        );
+                    duplicateClaimNames.Add(claim.Type);
                 }
             }
         }
 
+        // Remove any claimType detected to have more than one instance.
+        // For example: scp claim is a json array of strings. If the user has multiple scopes,
+        // the claimType will be present multiple times.
+        foreach (string claimType in duplicateClaimNames)
+        {
+            claimsInRequestContext.Remove(claimType);
```

Comment:
Doesn't simply ignoring the claims array misinform the constructed session context? Today, we have been erroring out, going forward if we dont want to error, I think better would be to still include these somehow in the session context.

Reply:
I see your point. That would be better, yes.
Another thought: the issue should be split into two separate work items: 
1. Enable compatibility with dupe claim types in DB Policies 
2. Enable compatibility with dupe claim types in session context. (Quicker with fewer cascading requirements/changes to make this work.)